ocaml/testsuite/tests/parsetree/source_jane_street.ml

```diff
@@ -4,14 +4,22 @@
    expect;
 *)
 
-(* This file is two tests in one! It's both an expect-test, run in the
-   normal way. It *also* is processed by [test.ml] (in this same directory)
-   to make sure that Pprintast round-trips. *)
-
 (* There should be examples of all Jane Street syntax here. This file can
    thus additionally serve as a cheap-and-cheerful documentation for all of
    our features. *)
 
+(* This file is three tests in one!
+    - It's an expect-test, run in the normal way.
+    - It is processed by [test.ml] (in this same directory) to make sure that Pprintast round-trips.
+    - It is processed by [test_ppx.ml] (in this same directory) to make sure that ppxes run properly
+      on examples of Jane Street syntax.
+*)
+
+(* Use [language-extensions/pprintast_unconditional.ml] to document pieces of syntax that
+   can't be added to this file (e.g. because you haven't yet implemented type-checking for
```

Comment:
Is the problem that test_ppx doesn't like tests that lead to `Misc.fatal_error`? Then we should say that more explicitly. Because features that don't type-check can still be tested fine in an expect-test -- you'll just expect type-checking to fail.

We could probably reasonably make one "unimplemented" error somewhere that any unimplemented features could use instead of `Misc.fatal_error` if that's indeed the problem.

Reply:
I thought `test_ppx` didn't like tests that lead to `Misc.fatal_error`, but it turns out that it's fine as long as there is a type error further up in the file.

I've moved the modal kind test into `source_jane_street.ml` and removed the confusing cross-linking of the files.

Reply:
I think this is an improvement. But now: why do we need `pprintast_unconditional`? Isn't the printing already covered in the new test?

Reply:
I think it's nice to test the individual `Pprintast` entry points somewhere  (`expression`, `tyvar`, `mode`, ...) against our extensions. `source_jane_street.ml` just tests `Pprintast.structure` and `Pprintast.signature`.